src/error.rs

```diff
@@ -60,32 +63,80 @@ impl Display for Error {
                 DeserializeBadOption => "Found an Option discriminant that wasn't 0 or 1",
                 DeserializeBadEnum => "Found an enum discriminant that was > u32::max_value()",
                 DeserializeBadEncoding => "The original data was not well encoded",
-                SerdeSerCustom => "Serde Serialization Error",
-                SerdeDeCustom => "Serde Deserialization Error",
+                SerdeSerCustom(message) =>
+                    return write!(f, "Serde Serialization Error: {}", message),
+                SerdeDeCustom(message) =>
+                    return write!(f, "Serde Deserialization Error: {}", message),
                 CollectStrError => "Error while processing `collect_str` during serialization",
             }
         )
     }
 }
 
+/// A custom error message
+///
+/// Only contains an actual message when the `alloc` feature is enabled, otherwise it is empty.
+#[derive(Eq, PartialEq, Debug, Clone)]
+pub struct Message {
+    #[cfg(feature = "alloc")]
+    message: alloc::string::String,
+}
+
+impl Display for Message {
+    fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        #[cfg(feature = "alloc")]
+        let result = f.write_str(&self.message);
+
+        #[cfg(not(feature = "alloc"))]
+        let result =
+            f.write_str("[Message not stored unless postcard is built with the `alloc` feature]");
+
+        result
+    }
+}
+
 /// This is the Result type used by Postcard.
 pub type Result<T> = ::core::result::Result<T, Error>;
 
 impl serde::ser::Error for Error {
-    fn custom<T>(_msg: T) -> Self
+    fn custom<T>(msg: T) -> Self
     where
         T: Display,
     {
-        Error::SerdeSerCustom
+        #[cfg(feature = "alloc")]
+        {
+            use core::fmt::Write;
+            let mut message = alloc::string::String::new();
+            write!(message, "{}", msg).expect("Write message to string");
```

Comment:
Hmm, I'm not particularly fond if this error message. It doesn't seem very expressive or useful. I'd say either unwrap or improve the error message to actually be useful.

Reply:
I just decided to unwrap it. I don't think that writing to a string is supposed to fail unless it's out-of-memory or something like that, and we'd still get a backtrace on the unwrap, so I don't think a message here is bringing us much benefit.